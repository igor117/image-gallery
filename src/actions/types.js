import { createActionTypes } from '../utils'

export const ITEM = createActionTypes('ITEM', [
  'GET_PHOTOS',
  'FAVOURITE',
  'SUCCESS',
  'FAILURE',
])

export default ITEM
