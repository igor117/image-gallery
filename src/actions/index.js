import { createAction } from '../utils'
import { ITEM } from './types'

export const item = {
  request: () => createAction(ITEM.GET, { fetching: true, success: false, error: null }),
  getPhotos: () => createAction(ITEM.GET_PHOTOS, { fetching: true, success: false, error: null }),
  favourite: (id) => createAction(ITEM.FAVOURITE, { id, fetching: true, success: false, error: null }),
  success: data => createAction(ITEM.SUCCESS, {
    ...data, fetching: false, success: true, error: null,
  }),
  failure: error => createAction(ITEM.FAILURE, { ...error, fetching: false, success: false }),
}

export default item
