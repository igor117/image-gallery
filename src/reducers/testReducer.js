import { fromJS, List } from 'immutable'
import { ITEM } from '../actions/types'

const initialState = fromJS({
  data: new List([]),
  fetching: false,
  success: false,
  error: null,
})

export default function testReducer(state = initialState, action) {
  switch (action.type) {
    case ITEM.GET_PHOTOS:
    case ITEM.SUCCESS:
    case ITEM.FAILURE:
      return state.merge(action.payload)
    case ITEM.FAVOURITE:
      const { id } = action.payload;
      let data = state.get('data').slice(0);
      if (typeof data[id].favourite === 'undefined')
        data[id].favourite = true;
      else
        data[id].favourite = !data[id].favourite;
      
      return state.set('data', data);
    default:
      return state
  }
}
