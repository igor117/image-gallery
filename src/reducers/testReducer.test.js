import { fromJS, List } from 'immutable'
import { ITEM } from '../actions/types'
import testReducer from './testReducer';
describe('testReducer', () => {
  const initialState = fromJS({
    data: new List([]),
    fetching: false,
    success: false,
    error: null,
  })
  it('returns the initial state when an action type is not passed', () => {
    const reducer = testReducer(undefined, {});
    expect(reducer).toEqual(initialState);
  });

  it('should handle GET_PHOTOS', () => {
    expect(
      testReducer(initialState, {
        type: ITEM.GET_PHOTOS
      })
    ).toMatchSnapshot()
  })

  it('should handle SUCCESS', () => {
    expect(
      testReducer(initialState, {
        type: ITEM.SUCCESS
      })
    ).toMatchSnapshot()
  })
});
