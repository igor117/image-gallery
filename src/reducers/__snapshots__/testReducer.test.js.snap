// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testReducer should handle GET_PHOTOS 1`] = `
Immutable.Map {
  "data": Immutable.List [],
  "fetching": false,
  "success": false,
  "error": null,
}
`;

exports[`testReducer should handle SUCCESS 1`] = `
Immutable.Map {
  "data": Immutable.List [],
  "fetching": false,
  "success": false,
  "error": null,
}
`;
